src/transformers/testing_utils.py

```diff
@@ -1570,6 +1572,34 @@ def LoggingLevel(level):
         transformers_logging.set_verbosity(orig_level)
 
 
+class TemporaryHubRepo:
+    """
+    This is a context manager to temporarily change transformers modules logging level to the desired value and have it
+    restored to the original setting at the end of the scope.
+
+    Example:
+
+    ```python
+    with LoggingLevel(logging.INFO):
+        AutoModel.from_pretrained("openai-community/gpt2")  # calls logger.info() several times
+    ```
+    """
+    def __init__(self, prefix="repo", random_id=None, token=None):
+        self.prefix = prefix
+        self.random_id = random_id
+        self.token = token
+        self.repo_id = f"{self.prefix}-{self.random_id}"
```

Comment:
(nit) be careful that this is a `repo_name` and not a `repo_id` per-se since it doesn't contain the namespace. Also since `self.random_id` can be `None`, then the repo_name might not be unique.